e2e_tests/tests/experiment/test_core.py

```diff
@@ -65,11 +66,11 @@ def test_metric_gathering() -> None:
         batches_trained = step["total_batches"]
 
     # Check validation metrics.
-    for step in trials[0]["steps"]:
-        validation = step["validation"]
-        metrics = validation["metrics"]
-        actual = metrics["validation_metrics"]
-        batches_trained = step["total_batches"]
+    validation_workloads = list(filter(lambda w: w.validation, trials[0].workloads))
```

Comment:
now that i've read more, these functions seem pretty generally useful. can we put them somewhere else?